### PR TITLE
Invoke

### PR DESCRIPTION
**Pull request by andresmgot**
Add support for the `invoke` method with the options:
```
    --function / -f (required) ......... The function name
    --path / -p ........................ Path to JSON or YAML file holding input data
    --log / -l ......................... Trigger logging data output
    --data / -d ........................ input data
```

### CODE REVIEW

File: invoke/kubelessInvoke.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+const BbPromise = require('bluebird');
+const path = require('path');
+const request = require('request');
+const helpers = require('../lib/helpers');
+
+class KubelessInvoke {
+  constructor(serverless, options) {
+    this.serverless = serverless;
+    this.options = options || {};
+    this.provider = this.serverless.getProvider('google');
```
Comment (tompizmor) [inline invoke/kubelessInvoke.js]: what is this google provider? should be like that by default?

File: invoke/kubelessInvoke.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            reject(new this.serverless.classes.Error(data.statusMessage, data.statusCode));
+          }
+          resolve(data);
+        }
+      });
+    });
+  }
+  log(response) {
+    if (this.options.log) {
+      console.log('--------------------------------------------------------------------');
```
Comment (tompizmor) [inline invoke/kubelessInvoke.js]: I guess this is needed to delimitate some output?

File: invoke/kubelessInvoke.js (hunk 1 above)
Comment (andresmgot) [inline invoke/kubelessInvoke.js]: Yes. For the moment it is not supported to set it as "kubeless": #2

File: invoke/kubelessInvoke.js (hunk 2 above)
Comment (andresmgot) [inline invoke/kubelessInvoke.js]: Yes, I wanted to mimic the behaviour from:
https://github.com/serverless/serverless/blob/bb7700dd0d30bbcf787129146ae48e33eb8de4f0/lib/plugins/aws/invoke/index.js#L114

**Comment by tompizmor** [comment]
LGTM

**Comment by sebgoa** [comment]
can you add tests, see:

https://github.com/serverless/serverless-openwhisk/blob/master/invoke/tests/index.js

https://github.com/serverless/serverless-google-cloudfunctions/blob/master/invoke/googleInvoke.test.js

**Comment by andresmgot** [comment]
> can you add tests, see:

Sure, it was in my roadmap. I can add them in the same PR

**Comment by sebgoa** [comment]
yes, I think it is better in the same pr

**Comment by andresmgot** [comment]
Done

**Comment by sebgoa** [comment]
@tompizmor can you test this PR locally with the latest kubeless installed in minikube ? then give a /lgtm ?

**Comment by tompizmor** [comment]
yes, I can test it

**Comment by tompizmor** [comment]
I tried this PR locally using minikube v0.20.0 and kubeless 0.0.14 and it works. However, it seems not to be working with the latest kubeless 0.0.15.

**Comment by andresmgot** [comment]
It seems to be an issue with kubeless 0.0.15 not with this code (`kubeless function ls` returns an exception). 

I am merging this PR. Let me know if you find something else.

**Comment by sebgoa** [comment]
+1 we are tracking the 0.0.15 issue

**Pull request merged**